internal/pkg/addon/package_test.go

```diff
@@ -23,12 +32,41 @@ func TestPackage(t *testing.T) {
 		bucket = "mockBucket"
 	)
 
+	lambdaZipHash := sha256.New()
+	indexZipHash := sha256.New()
+	indexFileHash := sha256.New()
+
+	// fs has the following structure:
+	//  .
+	//  ├─ lambda
+	//  │  ├─ index.js (contains lambda function)
+	//  ┴  └─ test.js (empty)
+	fs := afero.NewMemMapFs()
+	fs.Mkdir("/lambda", 0644)
+
+	f, _ := fs.Create("/lambda/index.js")
+	defer f.Close()
+	info, _ := f.Stat()
+	io.MultiWriter(lambdaZipHash, indexZipHash).Write([]byte("index.js " + info.Mode().String()))
+	io.MultiWriter(f, lambdaZipHash, indexZipHash, indexFileHash).Write([]byte(`exports.handler = function(event, context) {}`))
+
+	f2, _ := fs.Create("/lambda/test.js")
+	info, _ = f2.Stat()
+	lambdaZipHash.Write([]byte("test.js " + info.Mode().String()))
+
+	lambdaZipS3Path := fmt.Sprintf("manual/addons/mock-wl/assets/%s", hex.EncodeToString(lambdaZipHash.Sum(nil)))
+	indexZipS3Path := fmt.Sprintf("manual/addons/mock-wl/assets/%s", hex.EncodeToString(indexZipHash.Sum(nil)))
+	indexFileS3Path := fmt.Sprintf("manual/addons/mock-wl/assets/%s", hex.EncodeToString(indexFileHash.Sum(nil)))
```

Comment:
I see. Am I summarizing correctly that
1. These three variables are the "correct answer" that we want to check the hash against
2. `lambdaZipS3Path` is the correct hash for resources that wants to upload the whole `/lambda` folder
3. `indexZipS3Path` is the correct hash for resources that wants to zip and upload the `/lambda/index.js` file
4. `indexFileS3Path` is the correct hash for resources that want to just upload the `/lambda.index.js` file (without zipping)

Reply:
yeah, that's correct. these three paths should cover all potential cases of file hashes (folder, file, and zipped file)

Reply:
Nice! Nothing to address here - I just want to share an alternative:
```go
m.uploader.Expect.Upload(bucket, path).DoAndReturn(func(bucket string, path string) s3Location {
   // Compare got path with expected hash.
   require.Equal(indexHash, path)
})
```
This is more verbose and results in longer test files than what you have without making real difference. However I guess the one nice thing on its semantic message to other developers: it conveys the intention to "validate the hash instead of doing simple mocks" here.